src/test/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateUtilsTest.java

```diff
@@ -688,17 +689,21 @@ public void shouldCombineAllToolLocations() {
 
         PodTemplate podTemplate1 = new PodTemplate();
         List<ToolLocationNodeProperty> nodeProperties1 = new ArrayList<>();
-        nodeProperties1.add(new ToolLocationNodeProperty(new ToolLocationNodeProperty.ToolLocation("toolKey1@Test","toolHome1")));
+        ToolLocationNodeProperty toolHome1 = new ToolLocationNodeProperty(new ToolLocationNodeProperty.ToolLocation("toolKey1@Test", "toolHome1"));
+        nodeProperties1.add(toolHome1);
         podTemplate1.setNodeProperties(nodeProperties1);
 
         PodTemplate podTemplate2 = new PodTemplate();
         List<ToolLocationNodeProperty> nodeProperties2 = new ArrayList<>();
-        nodeProperties2.add(new ToolLocationNodeProperty(new ToolLocationNodeProperty.ToolLocation("toolKey2@Test","toolHome2")));
+        ToolLocationNodeProperty toolHome2 = new ToolLocationNodeProperty(new ToolLocationNodeProperty.ToolLocation("toolKey2@Test", "toolHome2"));
+        nodeProperties2.add(toolHome2);
         podTemplate2.setNodeProperties(nodeProperties2);
 
         PodTemplate result = combine(podTemplate1,podTemplate2);
 
-        assertThat(result.getNodeProperties(), hasItems(nodeProperties1.get(0),nodeProperties2.get(0)));
+        assertThat(podTemplate1.getNodeProperties(), contains(toolHome1));
+        assertThat(podTemplate2.getNodeProperties(), contains(toolHome2));
+        assertThat(result.getNodeProperties(), contains(toolHome1, toolHome2));
```

Comment:
Is this test reproducing the bug? I thought that the problem was that the parent template would grow endlessly, so we should have a test asserting that after combination the parent template does _not_ contain some property.

Reply:
Or that the size of the parent templates nodeProperties didn't change.

Reply:
First assertion failed with the previous code, since there was a side-effect on the first pod template.